libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -3363,16 +3375,26 @@ void IRGeneratorForStatements::revertWithError(
 	vector<ASTPointer<Expression const>> const& _errorArguments
 )
 {
+	bool needsAllocation = true;
+	if (optional<size_t> size = staticEncodingSize(_parameterTypes))
+		if (ranges::all_of(_parameterTypes, [](auto const* type) { return type && type->isValueType(); }))
```

Comment:
Just added the additional restriction of all error arguments being value types to make sure we don't run into weird static array arguments that end up calculating hashes in scratch space during ABI encoding. Value types alone are safe.